scripts/generator/graphql_generator/csharp/SDK/CartLineItems.cs.erb

```diff
@@ -213,13 +222,24 @@ namespace <%= namespace %>.SDK {
         /// cart.LineItems.AddOrUpdate(product, selectedOptions, 3);
         /// \endcode
         public void AddOrUpdate(Product product, Dictionary<string, string> selectedOptions, long? quantity = null, Dictionary<string, string> customAttributes = null) {
-            string variantId = VariantIdFromSelectedOptions(product, selectedOptions);
+            var variantId = VariantIdFromSelectedOptions(product, selectedOptions);
+            var variant = GetVariantById(product, variantId);
```

Comment:
For sure, I think I'd break that out into another PR as well, as a precursor, so that can be a straight refactor which shouldn't be affecting any tests.